### PR TITLE
Add Tooltip positioners support for Chart.js

### PR DESCRIPTION
**Pull request by vankooch**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Tooltip positioners](http://www.chartjs.org/docs/latest/configuration/tooltip.html#position-modes)
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -275,6 +278,10 @@ declare namespace Chart {
         borderWidth?: number;
     }
 
+    interface ChartTooltipsStaticConfiguartion {
+        positioners: any;
```
Comment (mernen) [inline types/chart.js/index.d.ts]: That’s a really vague definition, but it doesn’t seem like there’s much to work with…

Maybe something like this:

```ts
interface ChartTooltipsStaticConfiguration {
    positioners: {[mode: string]: ChartTooltipPositioner};
}

type ChartTooltipPositioner = (elements: any[], eventPosition: Point) => Point;

interface Point {
    x: number;
    y: number;
}
```

The type of `elements` (described as "Chart.Element[]") is completely undocumented, AFAICT, so think it’s better to leave as `any`. And the documentation example mentioning `this` also appears to be bogus.

File: types/chart.js/index.d.ts [hunk 2]
```diff
@@ -44,6 +44,9 @@ declare class Chart {
     static controllers: {
         [key: string]: any;
     };
+
+    // Tooltip Static Options
+    static Tooltip: Chart.ChartTooltipsStaticConfiguartion;
```
Comment (mernen) [inline types/chart.js/index.d.ts]: Typo: Configu**ar**tion

**Review by mernen — CHANGES_REQUESTED** [review]

File: types/chart.js/index.d.ts (hunk 1 above)
Comment (vankooch) [reply types/chart.js/index.d.ts]: Yes you are right we could improve the definition.

When I dump one element we get following structure:

```javascript
var ChartElement =
    {
        hidden: false,
        _chart: { ... },
        _datasetIndex: 0,
        _index: 1,
        _model: {
            "x": 74.31506588976988,
            "y": 546,
            "skip": false,
            "radius": 4,
            "pointStyle": "circle",
            "backgroundColor": "rgb(21, 172, 82)",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderWidth": 0,
            "tension": 0,
            "hitRadius": 1
        },
        _start: {},
        _view: {
            "x": 110.31506588976988,
            "y": 546,
            "skip": false,
            "radius": 4,
            "pointStyle": "circle",
            "backgroundColor": "rgb(21, 172, 82)",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderWidth": 0,
            "tension": 0,
            "hitRadius": 1
        },
        _xScale: { ... },
        _xScale: { ... },
    };
```

The properties with ```{ ... }``` have circular connections so I did not dump them. The only property which look promising is ```hidden```. But I would also stick to ```any```.

I'll push this changes for review.

**Review by mernen — CHANGES_REQUESTED** [review]
Could you please fix the typo in “Configuration”? Other than that, looks good.

**Review by mernen — APPROVED** [review]

**Comment by vankooch** [comment]
Well even when I fix the Travis-CI error. We are not finished yet. There is a problem left with:

```typescript
interface ChartTooltipOptions {
        enabled?: boolean;
        custom?(a: any): void;
        mode?: string;
        intersect?: boolean;
        backgroundColor?: ChartColor;
        titleFontFamily?: string;
        titleFontSize?: number;
        titleFontStyle?: string;
        titleFontColor?: ChartColor;
        titleSpacing?: number;
        titleMarginBottom?: number;
        bodyFontFamily?: string;
        bodyFontSize?: number;
        bodyFontStyle?: string;
        bodyFontColor?: ChartColor;
        bodySpacing?: number;
        footerFontFamily?: string;
        footerFontSize?: number;
        footerFontStyle?: string;
        footerFontColor?: ChartColor;
        footerSpacing?: number;
        footerMarginTop?: number;
        xPadding?: number;
        yPadding?: number;
        caretSize?: number;
        cornerRadius?: number;
        multiKeyBackground?: string;
        callbacks?: ChartTooltipCallback;
        filter?(item: ChartTooltipItem): boolean;
        itemSort?(itemA: ChartTooltipItem, itemB: ChartTooltipItem): number;
        position?: "average"|"nearest";
        caretPadding?: number;
        displayColors?: boolean;
        borderColor?: ChartColor;
        borderWidth?: number;
    }
```

We need to change the ```position``` type to string, so that we can use custom ones.

**Comment by vankooch** [comment]
Ready for merge

**Pull request merged**